### PR TITLE
Switch to .toml files for new Pkg3 Registry process

### PR DESCRIPTION
**Pull request by NHDaly**
- Add Project.toml, Manifest.toml
- Remove REQUIRE

I directly converted our version specifications from the REQUIRE file (ie kept them as their corresponding `>=` meaning), giving us:
https://github.com/NHDaly/Blink.jl/blob/2c6e94d392b13cf20c6a0f025d31e57b187c9776/Project.toml#L22-L26

But I think more likely than not, we probably want to switch these to be semver versions. But I guess we should reflect first on whether those versions are even up to date anymore? I _vaguely_ recall wanting to set some upper bound version requirements in the past and not being able to, but I don't think we have any such problems right now.
Any thoughts on what's best for this?

Thanks!

- [ ] This is a reminder that after we merge this and start the registration PR, we should ping @travigd to start a registration PR for [WebIO.jl](https://github.com/JuliaGizmos/WebIO.jl).

### CODE REVIEW

**Comment by NHDaly** [comment]
Oh and also, @pfitzseb, I think I remember that you are in the anti-manifest file camp? I think I am in the opposed group as well, since it will hide breakages in upstream packages from our CI unless we keep it up-to-date. I think I vote for removing it, but i wanted to upload it here just for posterity. :)

**Comment by travigd** [comment]
You'll need to change the TravisCI file.

```yaml
script:
  - $TESTCMD -e 'using Pkg; Pkg.build(); Pkg.test("Blink"; coverage=true)'
```

Manifest should be removed imo (it will just create git diff noise and serve no purpose).

**Comment by travigd** [comment]
WebIO PR: https://github.com/JuliaRegistries/General/pull/1473

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+Mux = "a975b10e-0019-58db-a62f-e48ff68538c9"
+Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
+Sockets = "6462fe0b-24de-5631-8697-dd941f90decc"
+WebIO = "0f1e0344-ec1d-5b48-a673-e5cf874b6c29"
+WebSockets = "104b5d7c-a370-577a-8038-80a2059c5097"
+
+[compat]
+JSExpr = "≥ 0.3"
+Lazy = "≥ 0.11.3"
+WebIO = "≥ 0.3"
```
Comment (pfitzseb) [inline Project.toml]: Should probably just use `"0.3, 0.4, 0.5, 0.6, 0.7, 0.8"` here.

File: Manifest.toml [hunk 2]
```diff
@@ -0,0 +1,286 @@
+# This file is machine-generated - editing it directly is not advised
```
Comment (pfitzseb) [inline Manifest.toml]: Yeah, let's not have this in the repo, imho.

File: Project.toml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+MacroTools = "1914dd2f-81c6-5fcd-8719-6d5c9610ff09"
+Mustache = "ffc61752-8dc7-55ee-8c37-f3e9cdd09e70"
+Mux = "a975b10e-0019-58db-a62f-e48ff68538c9"
+Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
+Sockets = "6462fe0b-24de-5631-8697-dd941f90decc"
+WebIO = "0f1e0344-ec1d-5b48-a673-e5cf874b6c29"
+WebSockets = "104b5d7c-a370-577a-8038-80a2059c5097"
+
+[compat]
+JSExpr = "≥ 0.3"
```
Comment (pfitzseb) [inline Project.toml]: Should probably be `"0.5"`.

File: Project.toml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+Lazy = "≥ 0.11.3"
+WebIO = "≥ 0.3"
+WebSockets = "≥ 1.0.1"
```
Comment (pfitzseb) [inline Project.toml]: Maybe `"1.5"`.

File: Project.toml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+Base64 = "2a0f44e3-6c83-55bd-87e4-b1978d98bd5f"
+BinDeps = "9e28174c-4ba2-5203-b857-d8d62c4213ee"
+Distributed = "8ba89e20-285c-5b6f-9357-94700520ee1b"
+JSExpr = "97c1335a-c9c5-57fe-bc5d-ec35cebe8660"
+JSON = "682c06a0-de6a-54ab-a142-c8b1cf79cde6"
+Lazy = "50d2b5c4-7a5e-59d5-8109-a42b560f39c0"
+Logging = "56ddb016-857b-54e1-b83d-db4d58db5568"
+MacroTools = "1914dd2f-81c6-5fcd-8719-6d5c9610ff09"
+Mustache = "ffc61752-8dc7-55ee-8c37-f3e9cdd09e70"
+Mux = "a975b10e-0019-58db-a62f-e48ff68538c9"
```
Comment (pfitzseb) [inline Project.toml]: Would probably be best to require Mux v0.7.

File: Project.toml [hunk 6]
```diff
@@ -0,0 +1,32 @@
+name = "Blink"
+uuid = "ad839575-38b3-5650-b840-f874b8c74a25"
+authors = []
+version = "0.10.2"
```
Comment (pfitzseb) [inline Project.toml]: Might make sense to directly jump to `0.11`, considering the rather big changes in regard to WebIO.

File: Project.toml (hunk 3 above)
Comment (NHDaly) [reply Project.toml]: Just to clarify, do you mean `"≥ 0.5"`, or literally just `"0.5"`?

**Comment by NHDaly** [comment]
> You'll need to change the TravisCI file.
> 
> ```yaml
> script:
>   - $TESTCMD -e 'using Pkg; Pkg.build(); Pkg.test("Blink"; coverage=true)'
> ```

Ah, thanks, good call! :) I've also changed the travis file for our docs build, which will hopefully work correctly as well.

> Manifest should be removed imo (it will just create git diff noise and serve no purpose).

👍 I removed the Manifest file as well. Thanks!

**Comment by NHDaly** [comment]
@travigd  actually, unfortunately it looks like the latest WebIO `v0.8.6` has broken Blink.

Can that version be fixed and maybe you can bump the breaking changes to be `v0.9` so that we can keep an upper bound on `[v0.3 - v0.8]` and still work?
Here's the log showing the breakage:
```julia
(Blink) pkg> test Blink
     ...
   Testing Blink tests passed

(Blink) pkg> update
  Updating registry at `~/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Resolving package versions...
 Installed WebIO ─ v0.8.6
  Updating `~/.julia/dev/Blink/Project.toml`
  [0f1e0344] ↑ WebIO v0.8.5 ⇒ v0.8.6
  Updating `~/.julia/dev/Blink/Manifest.toml`
  [cd3eb016] ↑ HTTP v0.8.2 ⇒ v0.8.3
  [0f1e0344] ↑ WebIO v0.8.5 ⇒ v0.8.6
  Building WebIO → `~/.julia/packages/WebIO/9qsL8/deps/build.log`

(Blink) pkg> build
  Building MbedTLS → `~/.julia/packages/MbedTLS/X4xar/deps/build.log`
  Building WebIO ──→ `~/.julia/packages/WebIO/9qsL8/deps/build.log`
  Building Blink ──→ `~/.julia/dev/Blink/deps/build.log`
 Resolving package versions...

(Blink) pkg> test Blink
   Testing Blink
 Resolving package versions...
ERROR: LoadError: Javascript error      Error: Cannot find module 'debug'
```

**Comment by NHDaly** [comment]
(and to clarify, i get the same issues even if I pin `HTTP` back to `v0.8.5`, so I think the issue is with WebIO):

```julia
(Blink) pkg> test Blink
   Testing Blink
 Resolving package versions...
ERROR: LoadError: Javascript error      Error: Cannot find module 'debug'
...
ERROR: Package Blink errored during testing

(Blink) pkg> status
Project Blink v0.11.0
    Status `~/.julia/dev/Blink/Project.toml`
...
  [cd3eb016] + HTTP v0.8.2 #6748814 (https://github.com/JuliaWeb/HTTP.jl.git) ⚲
...
  [0f1e0344] + WebIO v0.8.6
...
```

File: Project.toml (hunk 1 above)
Comment (pfitzseb) [reply Project.toml]: Hm, right, just `0.8` would probably be better. I really don't like not upper bounding deps, and we shouldn't claim compatibility with a (as of yet non-existant) WebIO v0.9.

File: Project.toml (hunk 3 above)
Comment (pfitzseb) [reply Project.toml]: I meant `"0.5"`, which is equivalent to `"^0.5"`, so we wouldn't claim compatibility with e.g. 0.6 (same below :)).

Comment (NHDaly) [reply Project.toml]: Done, sounds good. Thanks!

File: Project.toml (hunk 1 above)
Comment (NHDaly) [reply Project.toml]: Thanks, I've tightened all the versions to be caret specifiers. I think this sounds good.

**Comment by NHDaly** [comment]
Cool! Sorry this got dropped for two weeks.

I was hoping to finish it before I went on vacation all of last week, but didn't manage it.

Blink is now broken on master from the latest WebIO change; I've opened an issue about it here: https://github.com/JuliaGizmos/WebIO.jl/issues/317
Pending that fix, this PR should be good to go! :)

In the meantime I guess we could also change our compatability to more specifically upper-bound against `v0.8.6`?

**Comment by NHDaly** [comment]
Okay, I've just gone ahead and downgraded `WebIO` to `"0.7"` for now. Assuming that seems reasonable, this should be good to merge once the CI build succeeds. :)

**Comment by pfitzseb** [comment]
Alright, everything's green. Wanna merge + tag? :)

**Pull request merged**